connect-button/src/main/java/com/ifttt/connect/ui/AnalyticsManager.java

```diff
@@ -0,0 +1,266 @@
+package com.ifttt.connect.ui;
+
+import android.content.Context;
+import android.os.AsyncTask;
+import androidx.annotation.MainThread;
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+import androidx.annotation.WorkerThread;
+import androidx.work.ExistingWorkPolicy;
+import androidx.work.OneTimeWorkRequest;
+import androidx.work.WorkManager;
+import com.ifttt.connect.BuildConfig;
+import com.ifttt.connect.analytics.tape.ObjectQueue;
+import com.ifttt.connect.analytics.tape.QueueFile;
+import com.squareup.moshi.JsonAdapter;
+import com.squareup.moshi.Moshi;
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import okio.Buffer;
+import okio.BufferedSink;
+import okio.Okio;
+import android.os.Build.VERSION;
+
+/*
+ * This is the main class implementing analytics for the ConnectButton SDK. It is responsible for :
+ * 1. Providing different analytics tracking methods
+ * 2. A Queue for storing the events, including synchronous add, read and remove operations.
+ * 3. Setting up the WorkManager with a one time work request to schedule event uploads to server.
+ * Events are scheduled to be uploaded for every 5 events or when submitFlush is explicitly called by the ConnectButton class
+ * */
+final class AnalyticsManager {
+
+    private static AnalyticsManager INSTANCE = null;
+
+    @VisibleForTesting
+    ObjectQueue<AnalyticsEventPayload> queue;
+
+    private final WorkManager workManager;
+
+    private static boolean analyticsDisabled;
+
+    /*
+     * This lock is to ensure that only one queue operation - add, remove, or peek can be performed at a time.
+     * This will help make sure that items are not removed while we are adding them to the queue.
+     * */
+    private final Object queueLock = new Object();
+    private static final int MAX_QUEUE_SIZE = 1000;
+    private static final int FLUSH_QUEUE_SIZE = 5;
+
+    private static final String WORK_ID_QUEUE_FLUSH = "analytics_queue_flush";
+    private static final String QUEUE_FILE_NAME = "analytics-queue-file";
+
+    private AnalyticsManager(Context context) {
+        /*
+         * Try to create a file based queue to store the analytics event
+         * Use the in-memory queue as fallback
+         **/
+        Moshi moshi = new Moshi.Builder().build();
+
+        try {
+            File folder = context.getDir("analytics-disk-queue", Context.MODE_PRIVATE);
+            QueueFile queueFile = createQueueFile(folder);
+            queue = ObjectQueue.create(queueFile, new AnalyticsPayloadConverter(moshi));
+        } catch (IOException e) {
+            queue = ObjectQueue.createInMemory();
+        }
+
+        workManager = WorkManager.getInstance(context);
+        analyticsDisabled = AnalyticsPreferences.getAnalyticsTrackingOptOutPreference(context);
```

Comment:
Because this now is a global flag, we should not cache the value here as it would cause the existing AnalyticsManager to ignore the updated value.